actors/evm/src/interpreter/execution.rs

```diff
@@ -114,53 +99,46 @@ macro_rules! def_ins {
 macro_rules! def_ins_raw {
     ($ins:ident ($arg:ident) $body:block) => {
         #[allow(non_snake_case)]
-        fn $ins(p: *mut Self) -> Result<ControlFlow, StatusCode> {
+        fn $ins(p: *mut Self) {
             // SAFETY: macro ensures that mut pointer is taken directly from a mutable borrow, used once, then goes out of scope immediately after
             let $arg: &mut Self = unsafe { p.as_mut().unwrap() };
             $body
         }
     };
 }
 
-macro_rules! def_ins_primop {
+macro_rules! def_ins_intrinsic {
     ($ins:ident) => {
         def_ins_raw! {
             $ins (m) {
-                instructions::$ins(&mut m.state.stack)?;
-                Ok(ControlFlow::Continue)
+                match instructions::$ins(m) {
+                    Ok(_) => {
+                        m.pc += 1;
+                    },
+                    Err(err) => {
+                        m.exit = Some(err);
+                    }
+                }
             }
         }
     };
 }
 
-macro_rules! def_ins_push {
-    ($ins:ident) => {
-        def_ins_raw! {
-            $ins (m) {
-                m.pc += instructions::$ins(&mut m.state.stack, &m.bytecode[m.pc + 1..])?;
-                Ok(ControlFlow::Continue)
+macro_rules! try_ins {
+    ($ins:ident($m:ident) => ($res:ident) $body:block) => {
+        match instructions::$ins($m) {
+            Ok($res) => $body,
+            Err(err) => {
+                $m.exit = Some(err);
             }
```

Comment:
I'd be interested to see if the following were faster:

```rust
$m.exit = instructions::$ins($m).err()
$m.pc += 1; // when actually changing the PC.
```

In all cases. Basically, avoid the branches.

Of course, that might actually be _slower_ in terms of gas used. But I'm still curious.

Reply:
yeah, i will try it.

Reply:
err branches on its own perhaps? Or is it a vtable thing?

I am skeptical whether this would be any faster, and it is also somewhat harder to reason about.

Reply:
also, there is an assignment that doesnt need to happen 99.9% of the time; potentially just as bas as a correctly predicted branch or worse.